### PR TITLE
Video Import Command

### LINKED ISSUE #3: Video Import Script
## Story

Given I am a Developer
And I want to import a new video
When I execute `bake video import CHANNELSLUG youtube VIDEOID`
Then the video metadata* is added to `data/videos/CHANNELSLUG/VIDEOID.yml`

### Video MetaData

```
id: VIDEO_ID
title: VIDEO_TITLE.split(" | ").first
channel: CHANNELSLUG
series: VIDEO_TITLE.split(" | ").last
description: VIDEO_DESCRIPTION.paragraphs.first
source: youtube
```

### PR DESCRIPTION
**Pull request by jmickey**
Aims to close #3. API is a little different than usual, usage will be along the lines of:

```bash
bake import video --provider youtube --creator philosophytube --id kjdfnvskjdlnv123
```

This will be more inline with the intended state of the refactor once it is complete. 

Result:
image

### CODE REVIEW

File: util/channel.go [hunk 1]
```diff
@@ -317,6 +317,29 @@ type ChannelPage struct {
 	Videos []string `yaml:",omitempty"`
 }
 
+// GetChannelPage returns the ChannelPage for the specifid channel
+func (c *Channel) GetChannelPage(projectRoot string) *ChannelPage {
```
Comment (dirkkelly) [inline util/channel.go:321]: @jaymickey @j16r is the best practice in go to use full variable names?

```suggestion
func (channel *Channel) GetChannelPage(projectRoot string) *ChannelPage {
```

File: util/channel.go [hunk 2]
```diff
@@ -362,3 +385,31 @@ func CreateChannelPage(channel *Channel, projectRoot string) error {
 
 	return nil
 }
+
+// AddVideo adds a video to the channel page and saves it
+func (cp *ChannelPage) AddVideo(id, projectRoot string) error {
```
Comment (dirkkelly) [inline util/channel.go:390]:
```suggestion
func (channelPage *ChannelPage) AddVideo(id, projectRoot string) error {
```

File: util/channel.go [hunk 3]
```diff
@@ -362,3 +385,31 @@ func CreateChannelPage(channel *Channel, projectRoot string) error {
...
+	cp.Videos = append(cp.Videos, id)
+	err := cp.save(projectRoot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cp *ChannelPage) save(projectRoot string) error {
```
Comment (dirkkelly) [inline util/channel.go:399]:
```suggestion
func (channelPage *ChannelPage) save(projectRoot string) error {
```

File: cmd/video.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+var (
+	id       string
+	creator  string
+	provider string
+)
+
+func init() {
+	importRootCmd.AddCommand(videoCmd)
+
+	videoCmd.Flags().StringVar(&id, "id", "", "ID of the video, e.g. xspEtjnSfQA is the ID for https://www.youtube.com/watch?v=xspEtjnSfQA")
```
Comment (dirkkelly) [inline cmd/video.go:34]: Can we add support for full urls? That's what users are going to be submitting. Can be in a further refactor.

**Review by dirkkelly — APPROVED** [review]
This is awesome thank you! I have a build on my machine which I'll use for processing from now on. 

Huge timesaver thank you!

I have some questions but nothing that's blocking this being merged, will ask @j16r for his input.

File: util/channel.go (hunk 1 above)
Comment (jmickey) [reply util/channel.go:321]: In Go, for methods on a struct it's idiomatic for the underlying struct variable to be short, generally a single letter.  See an example here: https://github.com/gohugoio/hugo/blob/71b8d8b6a4127acacda8ada08cd61d7bfb18e962/navigation/menu.go#L74

Also see here from the official Golang style guide: https://github.com/golang/go/wiki/CodeReviewComments#receiver-names

File: util/channel.go (hunk 2 above)
Comment (jmickey) [reply util/channel.go:390]: Same as https://github.com/breadtubetv/bake/pull/15#discussion_r284642798

File: util/channel.go (hunk 3 above)
Comment (jmickey) [reply util/channel.go:399]: Same as https://github.com/breadtubetv/bake/pull/15#discussion_r284642798

File: cmd/video.go (hunk 4 above)
Comment (jmickey) [reply cmd/video.go:34]: Will add an issue for this!

**Pull request merged**